src/lib/Simplify.hs

```diff
@@ -40,17 +40,31 @@ simplifyModule scope (Module Core decls bindings) = do
   Module Simp (toNest declsNotDone) (bindings <> bindings')
 simplifyModule _ (Module ir _ _) = error $ "Expected Core, got: " ++ show ir
 
-evalSimplified :: Monad m => Module -> (Block -> m Atom) -> m Module
-evalSimplified (Module Simp Empty bindings) _ =
-  return $ Module Evaluated Empty bindings
-evalSimplified (Module Simp decls bindings) evalBlock = do
+splitSimpModule :: TopEnv -> Module -> (Block, Abs Binder Module)
```

Comment:
Yes, that would work too. I don't have strong feelings but I tend to prefer passing around data instead of functions where possible. I don't think we inspect its internals right now but we might want to, e.g. for `traceM` debugging.